djangosaml2idp/processors.py

```diff
@@ -47,14 +48,14 @@ def _get_nameid_opaque(cls, user_id: str, salt: bytes = b'', *args, **kwargs) ->
         return opaque.hexdigest()
 
     @classmethod
-    def get_nameid_persistent(cls, user_id: str, user: User, sp_entityid: str = '', idp_entityid: str = '') -> str:
+    def get_nameid_persistent(cls, user_id: str, sp: ServiceProvider, user: settings.AUTH_USER_MODEL, *args, **kwargs) -> str:
         """ Get PersistentID in TransientID format
             see: http://software.internet2.edu/eduperson/internet2-mace-dir-eduperson-201602.html#eduPersonTargetedID
         """
-        return '!'.join([idp_entityid, sp_entityid, cls._get_nameid_opaque(user_id, salt=str(user.pk).encode())])
+        return str(PersistentId.objects.get_or_create(sp=sp, user=user)[0].persistent_id)
```

Comment:
Does this not return a single entity? I've honestly never used it. Seems odd to have to do [0].

Reply:
I thought it was really weird too but it returns a tuple: https://docs.djangoproject.com/en/3.0/ref/models/querysets/#get-or-create

Reply:
Ah got it. Makes sense. Didn't read that far into the docs.